### PR TITLE
feature: updated self hosting guide

### PR DESCRIPTION
**Pull request by fastfists**

### CODE REVIEW

File: src/pages/self_hosting.mdx [hunk 1]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+```sh {{ title: 'Shell' }}
+cd arguflow
+cp .env.example .env
+docker compose up -d
+```
+
+## Step 4 - Configure Caddyfile
+Install caddy using the instructions here: https://caddyserver.com/docs/install
+
+We have an example Caddyfile in the root of the repository. You can use this as a starting point for your own configuration.
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:39]: I could not find this

File: src/pages/self_hosting.mdx [hunk 2]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+S3_ENDPOINT=http://s3:9000
+QDRANT_COLLECTION="vault"
+# Update with your own api keys
+SENDGRID_API_KEY=SG.******************************************************************
+OPENAI_API_KEY=sk-************************************************
+# You might want better passwords
+SECRET_KEY=01234012340123401234012340123401234012340123401234012340123401234012340123401234
+SALT="goodsaltisveryyummy"
+S3_USER="s3user"
+S3_PASSWORD="s3password"
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:85]: I have `ACCESS_KEY` and `SECRET_KEY`

Comment (skeptrunedev) [reply src/pages/self_hosting.mdx:85]: Looks like these are just missing

File: src/pages/self_hosting.mdx [hunk 3]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+LIBREOFFICE_PATH=libreoffice
+S3_ACCESS_KEY=ZaaZZaaZZaaZZaaZZaaZ
+S3_SECRET_KEY=ssssssssssssssssssssTTTTTTTTTTTTTTTTTTTT
+S3_BUCKET=vault
+VERIFICATION_SERVER_URL=http://127.0.0.1:8091/get_url_content
+QDRANT_API_KEY=qdrant_pass
+COOKIE_SECURE=false
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:92]: Probably should default this to `true`

File: src/pages/self_hosting.mdx [hunk 4]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+ALERT_EMAIL="developer@arguflow.gg"
+USE_EMBED_SERVER=0
+PARSER_COMMAND="./vault-nodejs/scripts/card_parser.js"
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:95]: Missing

File: src/pages/self_hosting.mdx [hunk 5]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+EMBEDDING_SERVER_CALL="http://localhost:5000/encode"
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:96]: Missing

File: src/pages/self_hosting.mdx [hunk 6]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+EMBEDDING_SIZE=1536
+
+# Only lines that need to get changed for now are the ones below
+PUBLIC_CHAT_URL=https://self-chat.arguflow.ai
+PUBLIC_SEARCH_URL=https://self-search.arguflow.ai
+API_HOST=https://self-api.arguflow.ai/api
+
+# You might want better passwords
+MINIO_ROOT_USER=rootuser
+MINIO_ROOT_PASSWORD=rootpassword
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:106]: Missing

File: src/pages/self_hosting.mdx [hunk 7]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+ALERT_EMAIL="developer@arguflow.gg"
+USE_EMBED_SERVER=0
+PARSER_COMMAND="./vault-nodejs/scripts/card_parser.js"
+EMBEDDING_SERVER_CALL="http://localhost:5000/encode"
+EMBEDDING_SIZE=1536
+
+# Only lines that need to get changed for now are the ones below
+PUBLIC_CHAT_URL=https://self-chat.arguflow.ai
+PUBLIC_SEARCH_URL=https://self-search.arguflow.ai
+API_HOST=https://self-api.arguflow.ai/api
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:102]: Missing

File: src/pages/self_hosting.mdx [hunk 8]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+``` {{ title: '.env.chat' }}
+VITE_API_HOST=https://self-api.arguflow.ai/api
+```
+
+``` {{ title: '.env.search' }}
+API_HOST=https://self-api.arguflow.ai/api
+PUBLIC_HOST=https://self-search.arguflow.ai
+PUBLIC_API_HOST=https://self-api.arguflow.ai/api
+```
+
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:119]: Is something missing here in terms of restarting the containers?

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

File: src/pages/self_hosting.mdx [hunk 9]
```diff
@@ -12,11 +12,122 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+# Define a site block for search.arguflow.ai
+self-search.arguflow.ai {
+    reverse_proxy localhost:3001
+}
+```
+
+Once you have this you can run the caddy daemon using systemctl
+```
+sudo systemctl start caddy.service
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:65]: Does this have to go after DNS for SSL cert creation?

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

File: src/pages/self_hosting.mdx [hunk 10]
```diff
@@ -12,11 +12,126 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+```
+
+
+## Step 7 - Profit 🦀🦀🦀
+
+Conngratualtions you have now self hosted the Arguflow API. You can now use the API by navigating to your domain name in your browser on either chat or search.
```
Comment (densumesh) [inline src/pages/self_hosting.mdx:137]: *Congratulations

File: src/pages/self_hosting.mdx [hunk 11]
```diff
@@ -12,11 +12,130 @@ This guide will demonstrate how to self host the Vault API on your own server or
...
+Once you have this you can run the caddy daemon using systemctl
+```
+sudo systemctl enable caddy.service
+sudo systemctl restart caddy.service
+```
+
+
+## Step 7 - Profit 🦀🦀🦀
+
+Conngratualtions you have now self hosted the Arguflow API. You can now use the API by navigating to your domain name in your browser on either chat or search.
```
Comment (skeptrunedev) [inline src/pages/self_hosting.mdx:141]:
```suggestion
Congratulations you have now self hosted the Arguflow API. You can now use the API by navigating to your domain name in your browser on either chat or search.
```

**Pull request merged**